### PR TITLE
Edit course titles

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
We might want to edit course lesson/module/year titles. They are what we use for links and what appears at the top of the page.

### Changes proposed in this pull request
1. Refactor preview mechanism to be more scalable dev-wise
2. Add editing titles.

### Guidance to review
Looking at one commit is pretty much like looking at them all - I copy pasted the solution from lessons to modules and years. Feel free to check that I didn't make a silly error somewhere.

### Testing
1. The refactor shouldn't change existing behaviour. To that end I added a check in our test that makes sure we don't change content of the object in db when we preview it.
2. Added a request test for each object that we have.

### CODE REVIEW

**Comment by amazimbe** [comment]
@kicferk1 did you branch off the other pr you had earlier? I noticed some of the changes that were on that pr are in here and it's not clear to me what's going on.

**Comment by kicferk1** [comment]
> @kicferk1 did you branch off the other pr you had earlier? I noticed some of the changes that were on that pr are in here and it's not clear to me what's going on.

Oh, I think you are right. My bad!

**Review by rupollbabu — COMMENTED** [review]
Unable to sign in with any role to modify/edit the course titles, Could you please check error logs.

File: spec/requests/core_induction_programme/course_lesson_spec.rb [hunk 1]
```diff
@@ -21,20 +22,27 @@
...
-    end
-    describe "PUT /core-induction-programme/years/:years/modules/:modules/lessons/:lessons" do
       it "redirects to the lesson page when saving content" do
-        put course_lesson_url, params: { commit: "Save changes", lesson_preview: "Adding new content" }
+        put course_lesson_url, params: { commit: "Save changes", content: "Adding new content" }
         expect(response).to redirect_to(course_lesson_url)
         get course_lesson_url
         expect(response.body).to include("Adding new content")
       end
+      it "redirects to the lesson page when saving title" do
```
Comment (seb-siem-softwire) [inline spec/requests/core_induction_programme/course_lesson_spec.rb:40]: Small space so it's easier to read

File: spec/requests/core_induction_programme/course_module_spec.rb [hunk 2]
```diff
@@ -11,32 +11,41 @@
...
-    end
-    describe "PUT /core-induction-programme/years/:years_id/modules/module_id" do
       it "redirects to the module page when saving content" do
-        put course_module_url, params: { commit: "Save changes", module_preview: "Adding new content" }
+        put course_module_url, params: { commit: "Save changes", content: "Adding new content" }
         expect(response).to redirect_to(course_module_url)
         get course_module_url
         expect(response.body).to include("Adding new content")
       end
+      it "redirects to the module page when saving title" do
```
Comment (seb-siem-softwire) [inline spec/requests/core_induction_programme/course_module_spec.rb:43]: Small space so it's easier to read

File: spec/requests/core_induction_programme/course_year_spec.rb [hunk 3]
```diff
@@ -11,32 +11,41 @@
...
-    end
-    describe "PUT /core-induction-programme/years/years_id" do
       it "redirects to the year page and updates content when saving changes" do
-        put course_year_url, params: { commit: "Save changes", year_preview: "Adding new content" }
+        put course_year_url, params: { commit: "Save changes", content: "Adding new content" }
         expect(response).to redirect_to(course_year_url)
         get course_year_url
         expect(response.body).to include("Adding new content")
       end
+      it "redirects to the year page when saving title" do
```
Comment (seb-siem-softwire) [inline spec/requests/core_induction_programme/course_year_spec.rb:43]: Small space

**Review by rupollbabu — COMMENTED** [review]
@kicferk1 -  Unable sign in(admin role) to view and edit course titles, Could you please check the error logs.

File: spec/requests/core_induction_programme/course_year_spec.rb (hunk 3 above)
Comment (ethanmills) [reply spec/requests/core_induction_programme/course_year_spec.rb:43]:
```
commit fd8be48dab3db80d0a3175ab91c5752adf8e5fab
Author: Ethan Mills <ethan.mills@softwire.com>
Date:   Tue Jan 26 16:24:24 2021 +0000

    ECF-RP-115: Michal hates whitespace
```
knew it...

Comment (kicferk1) [reply spec/requests/core_induction_programme/course_year_spec.rb:43]: 😂

**Review by amazimbe — APPROVED** [review]

**Comment by kicferk1** [comment]
> @kicferk1 - Unable sign in(admin role) to view and edit course titles, Could you please check the error logs.

I think it should be working now :)

**Review by rupollbabu — APPROVED** [review]

**Pull request merged**